searx/engines/cargo.py

```diff
@@ -0,0 +1,67 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""hex.pm"""
+
+from urllib.parse import urlencode
+from dateutil import parser
+from flask_babel import gettext
+
+
+about = {
+    "website": "https://crates.io/",
+    "wikidata_id": None,
+    "official_api_documentation": "https://crates.io/data-access",
+    "use_official_api": True,
+    "require_api_key": False,
+    "results": "JSON",
+}
+
+categories = ["it", "packages"]
+
+
+# engine dependent config
+paging = True
+search_url = "https://crates.io/api/v1/crates"
+
+
+def request(query: str, params):
+    args = urlencode({"page": params["pageno"], "q": query, "per_page": 20})
```

Comment:
per_page could be made configurable in `settings.yml` by making it a global variable called `page_size`, but that's not mandatory